fieldsolver/derivatives.cpp

```diff
@@ -270,60 +252,36 @@ void calculateDerivatives(
       dPerB->at(fsgrids::dperb::dPERBzdxy) = 0.0;
    } else {
       // Calculate xy mixed derivatives:
-      if ((sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY) || (sysBoundaryLayer == 1)) {
-         
+      if (sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY) {
```

Comment:
Three like this as the if of which we are in the else checks for sysBoundaryLayer == 1.